app/src/screens/document/ConfirmBelongingScreen.tsx

```diff
@@ -37,16 +38,57 @@ type ConfirmBelongingScreenProps = StaticScreenProps<Record<string, never>>;
 
 const ConfirmBelongingScreen: React.FC<ConfirmBelongingScreenProps> = () => {
   const selfClient = useSelfClient();
+  const [documentMetadata, setDocumentMetadata] = useState<{
+    documentCategory?: DocumentCategory;
+    signatureAlgorithm?: string;
+    curveOrExponent?: string;
+  }>({});
   const { trackEvent } = selfClient;
   const navigate = useHapticNavigation('Loading', {
-    params: {},
+    params: {
+      documentCategory: documentMetadata.documentCategory,
+      signatureAlgorithm: documentMetadata.signatureAlgorithm,
+      curveOrExponent: documentMetadata.curveOrExponent,
+    },
   });
   const [_requestingPermission, setRequestingPermission] = useState(false);
   const setFcmToken = useSettingStore(state => state.setFcmToken);
 
   useEffect(() => {
     notificationSuccess();
-  }, []);
+
+    const initializeProving = async () => {
+      try {
+        const selectedDocument = await loadSelectedDocument(selfClient);
+        if (selectedDocument?.data?.documentCategory === 'aadhaar') {
+          setDocumentMetadata({
```

Comment:
nit: we could use the if/else to set the object and call `setDocumentMetadata` once